src/main/java/com/bazel_diff/BazelSourceFileTarget.java

```diff
@@ -20,24 +19,24 @@
 
     BazelSourceFileTargetImpl(String name, byte[] digest, Path workingDirectory) throws IOException, NoSuchAlgorithmException {
         this.name = name;
-        byte[] data = null;
+        MessageDigest finalDigest = MessageDigest.getInstance("SHA-256");
         if (workingDirectory != null && name.startsWith("//")) {
             String filenameSubstring = name.substring(2);
             String filenamePath = filenameSubstring.replaceFirst(":", "/");
-            File sourceFile = new File(workingDirectory.toString(), filenamePath);
-            if (sourceFile.isFile() && sourceFile.canRead()) {
-                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-                outputStream.write(Files.readAllBytes(sourceFile.toPath()));
-                outputStream.write(digest);
-                data = outputStream.toByteArray();
-                outputStream.close();
-            }
-        } else {
-            data = digest;
-        }
-        MessageDigest finalDigest = MessageDigest.getInstance("SHA-256");
-        if (data != null) {
-            finalDigest.update(data);
+            String absoluteFilePath = workingDirectory.toString() + filenamePath;
```

Comment:
Ah no sorry, just notice that it's wrong here. When I run it in a test project, I pass working directory with trailing slash so it's fine.
But when I run in the actual project, working directory is passed without trailing slash and cause this `absoluteFilePath` points to something that does not exist, and that's why it's that fast! Because it read the filename only!

Reply:
I've another run, it's becomes 20% faster instead of 10x faster. Sorry 😞